homeassistant/components/teslemetry/entity.py

```diff
@@ -73,79 +129,81 @@ async def wake_up_if_asleep(self) -> None:
                         raise HomeAssistantError("Could not wake up vehicle")
                     await asyncio.sleep(times * 5)
 
-    def get(self, key: str | None = None, default: Any | None = None) -> Any:
-        """Return a specific value from coordinator data."""
-        return self.coordinator.data.get(key or self.key, default)
-
-    def set(self, *args: Any) -> None:
-        """Set a value in coordinator data."""
-        for key, value in args:
-            self.coordinator.data[key] = value
-        self.async_write_ha_state()
+    async def handle_command(self, command) -> dict[str, Any]:
+        """Handle a vehicle command."""
+        result = await super().handle_command(command)
+        if (response := result.get("response")) is None:
+            if message := result.get("error"):
+                # No response with error
+                LOGGER.info("Command failure: %s", message)
+                raise HomeAssistantError(message)
```

Comment:
Don't both log (above debug level) and raise an exception. It's enough to pass the message to the exception instance. The message will be logged by the exception handler.

If you want more information than the message for debugging, log that at debug level.

Reply:
Addressed in https://github.com/home-assistant/core/pull/117336